teammapper-frontend/mmp/src/map/handlers/nodes.ts

```diff
@@ -664,68 +689,141 @@ export default class Nodes {
+        };
+    
+        let siblings = params.getSiblings();
+    
+        if (nodeParent && params.isRoot) {
+            const [leftNodes, rightNodes] = siblings.reduce<[(Node | ExportNodeProperties)[], (Node | ExportNodeProperties)[]]>(
+                (acc, sibling) => {
+                    params.getOrientation(sibling) 
+                        ? acc[0].push(sibling) 
+                        : acc[1].push(sibling);
+                    return acc;
+                },
+                [[], []]
+            );
+    
             if (leftNodes.length <= rightNodes.length) {
-                coordinates.x -= 200
-                siblings = leftNodes
+                coordinates.x -= NODE_HORIZONTAL_SPACING;
+                siblings = leftNodes;
             } else {
-                coordinates.x += 200
-                siblings = rightNodes
+                coordinates.x += NODE_HORIZONTAL_SPACING;
+                siblings = rightNodes;
             }
         } else if (!node.detached) {
-            if (node.parent && this.getOrientation(node.parent)) {
-                coordinates.x -= 200
+            if (nodeParent && params.getOrientation(nodeParent)) {
+                coordinates.x -= NODE_HORIZONTAL_SPACING;
             } else {
-                coordinates.x += 200
+                coordinates.x += NODE_HORIZONTAL_SPACING;
             }
         }
-
+    
         if (siblings.length > 0) {
-            const lowerNode = this.getLowerNode(siblings)
-            coordinates.y = lowerNode.coordinates.y + 60
+            const lowerNode = 'isRoot' in node 
+                ? this.getNodeLowerNode(siblings as Node[])
+                : this.getExportLowerNode(siblings as ExportNodeProperties[]);
+            coordinates.y = (lowerNode?.coordinates?.y ?? 0) + NODE_VERTICAL_SIBLING_OFFSET;
         } else if (!node.detached) {
-            coordinates.y -= 120
+            coordinates.y -= NODE_VERTICAL_SPACING;
         }
+    
+        return coordinates;
+    }
 
-        return coordinates
+    /**
+     * Existing method to calculate the coordinates of "real", saved nodes in the database.
+     * This method will pass on existing methods such as this.getSiblings() to calculateNodeCoordinates, so existing implementations don't break
+     * @param node 
+     * @returns 
+     */
+    private calculateCoordinates(node: Node): Coordinates {
+        return this.calculateNodeCoordinates(
+            node,
+            {
+                getParent: () => node.parent,
+                getSiblings: () => this.getSiblings(node),
+                isRoot: node.parent?.isRoot ?? false,
+                getOrientation: (n: Node) => this.getNodeOrientation(n)
+            }
+        );
+    }
+
+    public applyCoordinatesToMapSnapshot = (map: MapSnapshot): MapSnapshot => {
+        const rootNode = map.find(x => x.isRoot);
+        
+        map.forEach(node => {
```

Comment:
`forEach` is used here on purpose, to work on the existing map and not create a new map e.g. via `map`? If possible, I would prefer creating a new map here - but maybe you need to update all the references.

Reply:
This was previously `map.map`, I don't know why I changed it. This returns a new map now and works properly.